src/main/java/org/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -247,8 +250,18 @@ private void navigateToNextEntry() {
         entryEditorTabs.add(new OtherFieldsTab(databaseContext, libraryTab.getSuggestionProviders(), undoManager, dialogService, preferencesService, stateManager, Globals.entryTypesManager, ExternalFileTypes.getInstance(), Globals.TASK_EXECUTOR, Globals.journalAbbreviationRepository));
 
         // General fields from preferences
+        int i = 0;
         for (Map.Entry<String, Set<Field>> tab : entryEditorPreferences.getEntryEditorTabList().entrySet()) {
-            entryEditorTabs.add(new UserDefinedFieldsTab(tab.getKey(), tab.getValue(), databaseContext, libraryTab.getSuggestionProviders(), undoManager, dialogService, preferencesService, stateManager, Globals.entryTypesManager, ExternalFileTypes.getInstance(), Globals.TASK_EXECUTOR, Globals.journalAbbreviationRepository));
+            String key = tab.getKey();
+            if (key.equals(preferencesService.getDefaults().get(JabRefPreferences.CUSTOM_TAB_NAME + "_def" + i))) {
```

Comment:
I'm not a big fan that the constants like CUSTOM_TAB_NAME are exposed. They are an implementation detail and shouldn't exposed outside. So for example, here you could add the default tab names to the entryEditorPreferences.

But actually I'm not sure why you need to check if its a default tab name in the first place. I think it's a nice touch that if the user chooses a name that we happen to have localized, why not always show the localized string. Simple example: the user decides to switch the abstract and the comment tab.

Reply:
Hi!
> I'm not a big fan that the constants like CUSTOM_TAB_NAME are exposed.

I agree it would be better to have that directly in the Preferences Objects. This goes for the keys and the default values. So each Preference Object would have it's own keys and it's own defaults-Map for those Strings. @calixtus argues this makes the list of preferences harder to implement.

> But actually I'm not sure why you need to check if its a default tab name in the first place.

Some background on this. We had some preferences that need localization. So for example the Entry-Editor-Tab name 'Abstract'. The default for the preference is 'Abstract'. JabRef should show 'Abstract' for english and 'Zusammenfassung' for german. Now we can think of a couple of cases:

When loading:
-  The preference that was loaded says 'Abstract'. We need to localize that since the language may not be set to english
-  The preference says something else. We don't localize this because it was changed by the user.

When storing:
-  JabRef is set to english. The new value to store says 'Abstract'. We can store that.
- JabRef is set to german. The new value says 'Zusammenfassung'. This was localized. We actually want to store 'Abstract' so it can be correctly localized next time it is needed (also if the language was changed in the meantime).

> why not always show the localized string

When the user changes the name, we cannot localize it anymore.

Reply:
Also see [ADR0023](https://github.com/JabRef/jabref/blob/8b9544737c123233d0756894bb2f79aae5905ff8/docs/adr/0023-localized-preferences.md)

Reply:
> So each Preference Object would have it's own keys and it's own defaults-Map for those Strings.

Not necessarily. My proposal would be to have a variable defaultTabNames in the preference object, which is initialized to the value of  `getDefaults().get(JabRefPreferences.CUSTOM_TAB_NAME + "_def" + i))`  using a constructor argument when the preference object is initialized in JabRefPreferences. In this way, you can access the default outside of JabRefPreferences without needing to worry how the preferences are stored there.

> When the user changes the name, we cannot localize it anymore.

We can still try to localize it, or not? So if the user puts "Abstract" as a custom tab name (assume it's not in default for the moment), then when displaying the tab, we get the localized string, say "Zusamenfassung" if the language is set to german. If the user already puts "Zusammenfassung" in the preferences, then Localization.lang doesn't do anything and returns "Zusammenfassung" again. There might be very rare cases where the user puts in a word in some language, which is a valid english word with a different translation (artificial example: Chinese user puts yue (moon) in, which is a valid english word (synonymy for Cantonese), so that Jabref would show Guǎngdōng huà (Cantonese in Chinese)). But these cases should be super rare with our limited dictionary. Would make the implementation a lot easier and might actually be what the user hopes to get